### PR TITLE
Styling on the task pages

### PR DESCRIPTION
**Pull request by victorbalan**
- Make the code editor relative to the body
- Refactor all the code tasks to use the
  created general-code-task
- Introduces:
  - general-code-task

### CODE REVIEW

**Comment by victorbalan** [comment]
#41 - Make the code editor more responsive

File: app/elements/editor-console/code-editor.html [hunk 1]
```diff
@@ -9,7 +15,7 @@
 			content-type="application/json"
 			>
 		</iron-ajax>
-		<div id="editor" style="height: 850px; z-index:0;"></div>
+		<div id="editor" style="height: 100%; z-index:0;"></div>
```
Comment (lorenzleutgeb) [inline app/elements/editor-console/code-editor.html]: Please get rid of this inline style.

File: app/elements/editor-console/general-console.html [hunk 2]
```diff
@@ -20,17 +15,14 @@
...
-					<content></content>
-				</div>
-			</iron-pages>
-		</div>
-
+		<iron-pages id="consoleOutput" selected="{{selected}}">
+			<div>
+				<p id="firstTab">{{text}}</p>
+			</div>
+			<div style="height: 100%">
```
Comment (lorenzleutgeb) [inline app/elements/editor-console/general-console.html]: Same thing. Please avoid inline styling.

File: app/elements/editor-console/javaut/javaut-console.html [hunk 3]
```diff
@@ -1,31 +1,37 @@
...
-    });
-  </script>
+	<style>
+		:host{
+			display: block;
+			height: 100%;
+		}
+	</style>
+	<template>
+		<general-console id="generalConsole" style="overflow: auto;">
```
Comment (lorenzleutgeb) [inline app/elements/editor-console/javaut/javaut-console.html]: No inline styling please.

File: app/elements/editor-console/output-test/output-test-console.html [hunk 4]
```diff
@@ -1,65 +1,58 @@
...
+		}
+		.green{
+			color: green;
+		}
+		p{
+			font-family: "Lucida Console", Monaco, monospace;
+		}
+	</style>
+	<template>
+		<general-console id="generalConsole" style="overflow: auto;">
```
Comment (lorenzleutgeb) [inline app/elements/editor-console/output-test/output-test-console.html]: Please get rid of inline styling.

File: app/elements/submission/challenge-page.html [hunk 5]
```diff
@@ -1,31 +1,37 @@
...
-    });
-  </script>
+	<style>
+		:host{
+			display: block;
+			height: 100%;
+		}
+	</style>
+	<template>
+		<div id="challenge" style="height: 100%;">
```
Comment (lorenzleutgeb) [inline app/elements/submission/challenge-page.html]: No inline styling!

**Comment by victorbalan** [comment]
Fixed

**Comment by victorbalan** [comment]
rebased

**Pull request merged**